### PR TITLE
Add the option to use background and threshold options on trim

### PR DESCRIPTION
**Pull request by tstm**
The trim function in libvips accepts optional parameters Background and Threshold, which can be quite important when trimming an image. The background is used to choose the color to actually trim away, and threshold is the amount of difference allowed, see [here](https://github.com/jcupitt/libvips/blob/f6db350a1e56e258e0a4e6ad6ee33b730b93b1c5/libvips/arithmetic/find_trim.c#L239-L274)

This should add those parameters, directly from the Options struct just as other such parameters.

One extra thing to add would be to have an option to detect the color to be used from ie. the top left pixel, to work with previously unknown images. This is also a feature I need.

### CODE REVIEW

**Comment by tstm** [comment]
Is there some way I could help or further things along to get this merged?

**Comment by h2non** [comment]
Sorry for the latency here. Ping me if I do not answer a PR in a few days.

File: vips.go [hunk 1]
```diff
@@ -503,15 +503,18 @@ func vipsSmartCrop(image *C.VipsImage, width, height int) (*C.VipsImage, error)
...
 	top := C.int(0)
 	left := C.int(0)
 	width := C.int(0)
 	height := C.int(0)
 
-	err := C.vips_find_trim_bridge(image, &top, &left, &width, &height)
+	err := C.vips_find_trim_bridge(image,
+		&top, &left, &width, &height,
+		C.double(background.R), C.double(background.G), C.double(background.B),
+		C.double(threshold))
```
Comment (greut) [inline vips.go]: `float32` and `C.double` ?

Comment (tstm) [reply vips.go]: Oh no! You're right. Should be float64 here, it's used as a double in libvips.

**Pull request merged**